pkg/provision/zdb.go

```diff
@@ -48,38 +48,41 @@ type ZDBResult struct {
 // ZDBMapping is a helper struct that allow to keep
 // a mapping between a 0-db namespace and the container ID
 // in which it lives
-type ZDBMapping struct {
+type zdbMapping struct {
 	m map[string]string
 	sync.RWMutex
 }
 
+// NewZDBMapping creates a new ZDBMapping object
+func NewZDBMapping() ZDBMapping {
+	return &zdbMapping{
+		m: make(map[string]string),
+	}
+}
+
 // Get returns the container ID where namespace lives
 // if the namespace is not found an empty string and false is returned
-func (z *ZDBMapping) Get(namespace string) (string, bool) {
+func (z *zdbMapping) Get(namespace string) (string, bool) {
 	z.RLock()
 	defer z.RUnlock()
 
-	if z.m == nil {
```

Comment:
I do like to have struct that have a usable default value. So I would keep this so someone creating a zdbMapping using a literal notation still have a working object.

Reply:
The can't create it because `zdbMapping` is not longer exposed, and they have to use the `NewZDBMapping` method to create one.

Reply:
That is only true for other package, provision package can still create it.

Reply:
in that case, developer of `provision` package should be aware of the implementation. I would rather have one initialization at the `new` method, instead of a check on each method call (even if there is not gonna be many of them)

Reply:
> in that case, developer of provision package should be aware of the implementation.

Yes, but why just removing code that give you this kind of protection for free ?? This type has 2 method, so 100% of the case were covered. Now we are just introducing more change to miss-use the type